internal/controller/adapters/sveltos/serviceset_controller.go

```diff
@@ -906,6 +900,27 @@ func helmChartFromSpecOrRef(
 	return helmChart, nil
 }
 
+// mergeHelmOptions merges the values from the given source ServiceHelmOptions to the destination ServiceHelmOptions
+func mergeHelmOptions(src, dst *kcmv1.ServiceHelmOptions) {
+	if src == nil || dst == nil {
```

Comment:
If `src` is non-nil while `dst` is nil, then logically after merging, `dst` should become equal to `src`. In the current implementation we simply return without modifying `dst`. I just want to confirm that this behavior is intentional.